src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -54,22 +54,29 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
             if (signaturesOnly)
                 return null;
 
+            var assemblyName = symbol.ContainingAssembly.Identity.Name;
+
+            _logger?.Clear();
+            _logger?.Log(FeaturesResources.Navigating_to_symbol_0_from_1, symbol, assemblyName);
+
             // If this is a reference assembly then we won't have the right information available, so bail out
             // TODO: find the implementation assembly for the reference assembly, and keep going: https://github.com/dotnet/roslyn/issues/55834
             var isReferenceAssembly = symbol.ContainingAssembly.GetAttributes().Any(attribute => attribute.AttributeClass?.Name == nameof(ReferenceAssemblyAttribute)
                 && attribute.AttributeClass.ToNameDisplayString() == typeof(ReferenceAssemblyAttribute).FullName);
             if (isReferenceAssembly)
+            {
+                _logger?.Log("");
                 return null;
+            }
 
             var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
 
             if (compilation.GetMetadataReference(symbol.ContainingAssembly) is not PortableExecutableReference { FilePath: not null and var dllPath })
             {
+                _logger?.Log(FeaturesResources.Source_is_a_reference_assembly);
```

Comment:
Did you mean to put this earlier? The message here should probably be different.